fdbserver/KeyValueStoreRocksDB.actor.cpp

```diff
@@ -407,20 +407,358 @@ class ReadIteratorPool {
+	void logStats(rocksdb::DB* db);
+	// PerfContext
+	void resetPerfContext();
+	void setPerfContext(int index);
+	void logPerfContext(bool ignoreZeroMetric);
+	// For Readers
+	Reference<Histogram> getReadRangeLatencyHistogram(int index);
+	Reference<Histogram> getReadValueLatencyHistogram(int index);
+	Reference<Histogram> getReadPrefixLatencyHistogram(int index);
+	Reference<Histogram> getReadRangeActionHistogram(int index);
+	Reference<Histogram> getReadValueActionHistogram(int index);
+	Reference<Histogram> getReadPrefixActionHistogram(int index);
+	Reference<Histogram> getReadRangeQueueWaitHistogram(int index);
+	Reference<Histogram> getReadValueQueueWaitHistogram(int index);
+	Reference<Histogram> getReadPrefixQueueWaitHistogram(int index);
+	Reference<Histogram> getReadRangeNewIteratorHistogram(int index);
+	Reference<Histogram> getReadValueGetHistogram(int index);
+	Reference<Histogram> getReadPrefixGetHistogram(int index);
+	// For Writer
+	Reference<Histogram> getCommitLatencyHistogram();
+	Reference<Histogram> getCommitActionHistogram();
+	Reference<Histogram> getCommitQueueWaitHistogram();
+	Reference<Histogram> getWriteHistogram();
+	Reference<Histogram> getDeleteCompactRangeHistogram();
+	// Stat for Memory Usage
+	void logMemUsagePerShard(std::string shardName, rocksdb::DB* db);
 
 private:
-	std::vector<std::tuple<const char*, int, std::vector<uint64_t>>> metrics;
+	// Global Statistic Input to RocksDB DB instance
+	std::shared_ptr<rocksdb::Statistics> stats;
+	// Statistic Output from RocksDB
+	std::vector<std::tuple<const char*, uint32_t, uint64_t>> tickerStats;
+	std::vector<std::pair<const char*, std::string>> propertyStats;
+	// Iterator Pool Stats
+	std::unordered_map<std::string, uint64_t> readIteratorPoolStats;
+	// PerfContext
+	std::vector<std::tuple<const char*, int, std::vector<uint64_t>>> perfContextMetrics;
+	// Readers Histogram
+	std::vector<Reference<Histogram>> readRangeLatencyHistograms;
+	std::vector<Reference<Histogram>> readValueLatencyHistograms;
+	std::vector<Reference<Histogram>> readPrefixLatencyHistograms;
+	std::vector<Reference<Histogram>> readRangeActionHistograms;
+	std::vector<Reference<Histogram>> readValueActionHistograms;
+	std::vector<Reference<Histogram>> readPrefixActionHistograms;
+	std::vector<Reference<Histogram>> readRangeQueueWaitHistograms;
+	std::vector<Reference<Histogram>> readValueQueueWaitHistograms;
+	std::vector<Reference<Histogram>> readPrefixQueueWaitHistograms;
+	std::vector<Reference<Histogram>> readRangeNewIteratorHistograms; // Zhe: haven't used?
+	std::vector<Reference<Histogram>> readValueGetHistograms;
+	std::vector<Reference<Histogram>> readPrefixGetHistograms;
+	// Writer Histogram
+	Reference<Histogram> commitLatencyHistogram;
+	Reference<Histogram> commitActionHistogram;
+	Reference<Histogram> commitQueueWaitHistogram;
+	Reference<Histogram> writeHistogram;
+	Reference<Histogram> deleteCompactRangeHistogram;
+
```

Comment:
Is it possible to also add the number of shards in the log? Or is there a convenience way to find that already?